metadata/src/lib.rs

```diff
@@ -106,19 +106,28 @@ where
     fn instantiate(
         &mut self,
         deps: &mut DepsMut,
-        _: &Env,
-        _: &MessageInfo,
+        env: &Env,
+        info: &MessageInfo,
         msg: Self::InstantiateMsg,
     ) -> Result<Response, Self::Error> {
         self.metadata.save(deps.storage, &msg.metadata)?;
-
-        Ok(Response::new())
+        let resp = Response::new().add_event(
+            Event::new("metadata-instantiate").add_attributes(
+                vec![
+                    ("contract_address", env.contract.address.to_string()),
+                    ("owner", info.sender.to_string()),
+                    ("metadata", serde_json::to_string(&msg.metadata).unwrap()),
```

Comment:
There are tons of unwraps in here, the whole repo could probably take a pass to get rid of them

Reply:
In general, if the code is a function that returns a Result, you should not unwrap. In tests it is fine, but in code where we want to know how or why things went wrong, unwrap loses that info